internal/pkg/cli/validate.go

```diff
@@ -209,6 +247,67 @@ func validateStorageType(val interface{}) error {
 	return fmt.Errorf(fmtErrInvalidStorageType, storageType, prettify(storageTypes))
 }
 
+func validateMySQLDBName(val interface{}) error {
+	const (
+		minMySQLDBNameLength = 1
+		maxMySQLDBNameLength = 64
+	)
+
+	dbName, ok := val.(string)
+	if !ok {
+		return errValueNotAString
+	}
+
+	// Check for db name length.
+	if len(dbName) < minMySQLDBNameLength || len(dbName) > maxMySQLDBNameLength {
+		return fmt.Errorf(fmtErrRDSNameBadSize, minMySQLDBNameLength, maxMySQLDBNameLength)
+	}
+
+	return validateDBNameCharacters(dbName)
+}
+
+func validatePostgreSQLDBName(val interface{}) error {
+	const (
+		minPostgreSQLDBNameLength = 1
+		maxPostgreSQLDBNameLength = 63
+	)
+
+	dbName, ok := val.(string)
+	if !ok {
+		return errValueNotAString
+	}
+
+	// Check for db name length.
+	if len(dbName) < minPostgreSQLDBNameLength || len(dbName) > maxPostgreSQLDBNameLength {
+		return fmt.Errorf(fmtErrRDSNameBadSize, minPostgreSQLDBNameLength, maxPostgreSQLDBNameLength)
+	}
+
+	return validateDBNameCharacters(dbName)
+}
+
+func validateDBNameCharacters(name string) error {
+	// Check for character constraints.
+	match := dbNameCharacterRegExp.FindStringSubmatch(name)
```

Comment:
Can we also check that there are no consecutive special characters? We should make sure that no bugs exist like what bit us on the stackset in #2005.

Reply:
Currently the db name is allowed to have consecutive underscores (hyphens and periods are not accepted characters so they shall not pass anyway) for both MySQL engine and PostgreSQL engine. Do we want to do the check for preemptive purpose?

Reply:
Not necessary. How about trailing underscores?

Also, how about periods and hyphens in the [logical storage name](https://github.com/aws/copilot-cli/pull/2123/files#diff-e08893c7bb37077621270debd504e87ad817fa7072e48ae9d100ebc2492f617dR118)?

Reply:
I also tried trailing underscores! It is allowed.

The storage name will be used as logical ID but they will go through `logicalIDSafe`. Hence the rds storage name here can have  -._ because they will be gone after `logicalIDSafe`.

LogicalID will then be used to construct DB Cluster identifier, which has a whole set of restrictions. However most of them can be satisfied for free because logical ID has only alphanumeric - except for one, that the identifier needs to start with a letter (as opposed to a number).

Hence the regexp - that it start with a letter, and the rest of the characters satisfy POSIX portable file name character set.